lib/logstash/filters/csv.rb

```diff
@@ -151,10 +151,14 @@ def filter(event)
   private
 
   # construct the correct Event field reference for given field_name, taking into account @target
-  # @param field_name [String] the bare field name without brakets
+  # @param field_name [String] the field name.
   # @return [String] fully qualified Event field reference also taking into account @target prefix
   def field_ref(field_name)
-    "#{@target}[#{field_name}]"
+    if field_name.start_with?("[")
```

Comment:
@jordansissel I think we need a test for the scenario you just described. I only see one for a nested `target` not for nested values in `columns`